src/libstd/sys/unix/process.rs

```diff
@@ -311,6 +312,18 @@ impl Process {
         if !envp.is_null() {
             *sys::os::environ() = envp as *const _;
         }
+
+        // Reset signal handling so the child process starts in a
+        // standardized state. We ignore SIGPIPE, and signal-handling
+        // libraries often set a mask; both of these get inherited,
+        // which most UNIX programs don't expect.
+        let mut set: c::sigset_t = mem::uninitialized();
+        if c::sigemptyset(&mut set) != 0 ||
+           c::pthread_sigmask(c::SIG_SETMASK, &set, ptr::null_mut()) != 0 ||
+           c::signal(libc::SIGPIPE, c::SIG_DFL) == c::SIG_ERR {
+            fail(&mut output);
+        }
```

Comment:
Aha! So back in 33a2191d0b880242b3bf9a32477a6b432f931c80 I was cleaning out some old code in the unix process spawning code, and I [removed a call to `rust_unset_sigprocmask`](https://github.com/rust-lang/rust/commit/33a2191d0b880242b3bf9a32477a6b432f931c80#diff-f05112808cca60914aac3cc823d54425L261) because there were no comments as to why it existed and I also had no idea why it existed. I think that this explains it, however!

To clarify, however, I've got a few questions:

* The [old implementation](https://github.com/rust-lang/rust/blob/33a2191d0b880242b3bf9a32477a6b432f931c80/src/rt/rust_builtin.c#L185-L191) used `sigprocmask` instead of `pthread_sigmask`, is there a reason that you opted to use the pthread version here instead? At this point there's only one thread, so there shouldn't be any multithreading worries.
* The old implementation also did nothing with `SIGPIPE` or other signals, so I'm somewhat wary here of explicitly setting it back to the default behavior. I can see how because our runtime explicitly resets this away from the default that it may be the only one we explicitly reset back, however, so I think I'm OK with that.